ansible/pbTestScripts/vagrantPlaybookCheck.sh

```diff
@@ -371,26 +371,33 @@ startVMPlaybookWin()
 
 destroyVM()
 {
-	local OS=$1
-	echo "Destroying the $OS Machine"
-	echo === `date +%T`: showing global status before pruning:
-	vagrant global-status
-	free
-	echo === showing global status while pruning:
-	vagrant global-status --prune
-	echo === Determining VM to destroy:
-	VM_TO_DESTROY=`vagrant global-status --prune | grep $OS | awk "/${gitFork}-${gitBranch}/ { print \\$1 }"`
-	if [ ! -z "$VM_TO_DESTROY" ]; then
-	  echo === Destroying VM with id $VM_TO_DESTROY
-	  vagrant destroy -f $VM_TO_DESTROY
+	if [[ "$retainVM" == false ]]; then
+		echo "Destroying the VM"
+		local OS=$1
+		echo "Destroying the $OS Machine"
+		echo === `date +%T`: showing global status before pruning:
+		vagrant global-status
+		free
+		echo === showing global status while pruning:
+		vagrant global-status --prune
+		echo === Determining VM to destroy:
+		VM_TO_DESTROY=`vagrant global-status --prune | grep $OS | awk "/${gitFork}-${gitBranch}/ { print \\$1 }"`
+		if [ ! -z "$VM_TO_DESTROY" ]; then
+		echo === Destroying VM with id $VM_TO_DESTROY
+		vagrant destroy -f $VM_TO_DESTROY
+		else
+		echo === NOT DESTROYING ANY VM as no suitable ID was found searching for $OS and ${gitFork}-${gitBranch}
+		fi
+			echo === Final status:
+			vagrant global-status --prune
+			free
 	else
-	  echo === NOT DESTROYING ANY VM as no suitable ID was found searching for $OS and ${gitFork}-${gitBranch}
+		echo "You have chosen to retain the VM. It will not be destroyed"
 	fi
-        echo === Final status:
-        vagrant global-status --prune
-        free
 }
 
+trap 'destroyVM $vagrantOS' EXIT
```

Comment:
Isn't vagrantOS potentially a list? That's why the original call to destroyVM is `$OS` as that's the loop iterator over `$vagrantOS` later in the script.

Reply:
It is potentially, but as far as I know it is never used as a list. Our VPC job only ever kicks off one OS at a time (separates multiple OSs over single jobs). But I'll make some adjustments